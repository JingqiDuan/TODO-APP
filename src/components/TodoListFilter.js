import React from 'react';
import {connect} from 'react-redux';
import {DateRangePicker} from 'react-dates';
import {setTextFilter, sortByDate,sortByImportance,setStartDate,setEndDate} from '../actions/filters';


class TodoListFilters extends React.Component{
    state = {
        calendarFocused: null
    };

    onFocusChange = (calendarFocused) =>{
        this.setState(()=>({calendarFocused}));
    };

    onDatesChange =({startDate,endDate})=>{
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    };

    render(){
        return(
            <div>
        <input 
        type="text" 
        value={this.props.filters.text} 
        onChange={ (e)=>{
            this.props.dispatch(setTextFilter(e.target.value));
        }}/>

        <select
            value = {this.props.filters.sortBy}
            onChange = {(e)=>{
                if(e.target.value==='amount'){
                    this.props.dispatch(sortByDate());
                }else if(e.target.value==='importance'){
                    this.props.dispatch(sortByImportance());
                }
            }}
        >
            <option value = "date">Date</option>
            <option value = "importance">Importance</option>
        </select>

        <DateRangePicker
            startDate = {this.props.filters.startDate}
            endDate = {this.props.filters.endDate}
            onDatesChange={this.onDatesChange}
            focusedInput = {this.state.calendarFocused}
            onFocusChange = {this.onFocusChange}
            showClearDates = {true}
            numberOfMonths ={1}
            isOutsideRange = {()=>false}
        />
            
    </div>
        );
    };
};




const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

export default connect(mapStateToProps)(TodoListFilters);