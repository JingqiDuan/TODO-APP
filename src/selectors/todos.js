import moment from 'moment';

//Get visible todos

//second argument is the breakdown of the filter
export default (todos,{text, sortBy,startDate,endDate})=>{
    return todos.filter((todo)=>{
        const createdAtMoment = moment(todo.createdAt);
        const startDateMatch = startDate ?startDate.isSameOrBefore(createdAtMoment,'day'):true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment,'day'): true;
        const textMatch = todo.eventName.toLowerCase().includes(text.toLowerCase())||
        todo.description.toLowerCase().includes(text.toLowerCase());

        return startDateMatch&&endDateMatch&&textMatch;
    }).sort((a,b)=>{
        if(sortBy ==='date'){
            return a.createdAt < b.createdAt?1:-1;
        }else if(sortBy ==='importance'){
            return a.importance<b.importance?1:-1;
        }
        });
};