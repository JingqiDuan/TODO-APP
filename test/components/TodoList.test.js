import React from 'react';
import {shallow} from 'enzyme';
import {TodoList} from '../../src/components/TodoList';
import todos from '../fixtures/todos';

test('should render TodoList with todos',()=>{
    const wrapper = shallow(<TodoList todos = {todos}/>);
    expect(wrapper).toMatchSnapshot();
});