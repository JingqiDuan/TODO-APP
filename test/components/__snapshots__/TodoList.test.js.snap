// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TodoList with todos 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoList
    todos={
        Array [
            Object {
              "createdAt": 0,
              "description": "",
              "eventName": "testEvent1",
              "id": "1",
              "importance": 7,
            },
            Object {
              "createdAt": -345600000,
              "description": "",
              "eventName": "testEvent2",
              "id": "2",
              "importance": 729,
            },
            Object {
              "createdAt": 345600000,
              "description": "",
              "eventName": "testEvent3",
              "id": "1",
              "importance": 7729,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Todo List
</h1>,
        Array [
          <TodoListItem
            createdAt={0}
            description=""
            eventName="testEvent1"
            id="1"
            importance={7}
/>,
          <TodoListItem
            createdAt={-345600000}
            description=""
            eventName="testEvent2"
            id="2"
            importance={729}
/>,
          <TodoListItem
            createdAt={345600000}
            description=""
            eventName="testEvent3"
            id="1"
            importance={7729}
/>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Todo List",
        },
        "ref": null,
        "rendered": "Todo List",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "createdAt": 0,
          "description": "",
          "eventName": "testEvent1",
          "id": "1",
          "importance": 7,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "createdAt": -345600000,
          "description": "",
          "eventName": "testEvent2",
          "id": "2",
          "importance": 729,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "createdAt": 345600000,
          "description": "",
          "eventName": "testEvent3",
          "id": "1",
          "importance": 7729,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Todo List
</h1>,
          Array [
            <TodoListItem
              createdAt={0}
              description=""
              eventName="testEvent1"
              id="1"
              importance={7}
/>,
            <TodoListItem
              createdAt={-345600000}
              description=""
              eventName="testEvent2"
              id="2"
              importance={729}
/>,
            <TodoListItem
              createdAt={345600000}
              description=""
              eventName="testEvent3"
              id="1"
              importance={7729}
/>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Todo List",
          },
          "ref": null,
          "rendered": "Todo List",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "createdAt": 0,
            "description": "",
            "eventName": "testEvent1",
            "id": "1",
            "importance": 7,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "createdAt": -345600000,
            "description": "",
            "eventName": "testEvent2",
            "id": "2",
            "importance": 729,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "createdAt": 345600000,
            "description": "",
            "eventName": "testEvent3",
            "id": "1",
            "importance": 7729,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
